### PR TITLE
Testing/eb email notification form 

### LINKED ISSUE #20: [🧑‍💻] Testing : Email Notification Form - E2E
- [x] - 😊  Should be able to input a first name, last name, email & select a state, language, and confirm subscription.
 - [x] - 😊  Should include text-based labels for each input.
 - [x] #29
 - [x] - 😊  Should be able to submit the form and see a message confirming successful subscription.
 - [x] - 😊  Should remove the missing input messages as soon as the user starts to fill it in.
 - [x] - 😊  Should be able to navigate the form with the tab key & submit the form using the enter key.
 - [x] - 😊  Should clear all inputs and selections once the form is submitted.
 - [x]  - 😊  Should be given a list of fifty states upon selecting the dropdown menu.
 - [x] - 😊  Should allow the user to change their email language preference.

 - [x] - ☹️ Should see a prompt if any of the required inputs are missing.
 - [x] - ☹️ Should see an error message if the email is already subscribed to notifications.
 - [x] - ☹️ Should see an error message if there is a problem with subscribing to email notifications.

### LINKED ISSUE #24: [🧑‍💻] Testing : Error & Loading Tests
😊  Should display a loading image while the submitting the form
☹️ Should display a an error message if the email is already subscribed
😊 Should display a message confirming the subscription to election notifications
☹️ Should display an error image & error message if the server can't complete the request
😊 Should display a button to send the user back to the form

### LINKED ISSUE #26: [⚙️] Refactor : Refactor Error Message Handling
When the submit button of the Email Notification Form is clicked, the user should see a success message that confirms they have been added to the database.

If the POST is **_unsuccessful_**, it could be the following reasons:
- Email address already exists in the database & this message should be displayed to the user under the form.
> -  response.ok
> -  status = 400
- Error creating a subscriber in the database for another reason (catch-all) & this message should be displayed to the user under the form.
> -  response.ok
> -  status = 400
- The server is down (500 error) & this should be displayed to the user with the Error component.
> -  !response.ok
> - status = 500 (??????) <--- this is a slippery slope...
- Internet connection has dropped & this error should be handled by the browser.

### LINKED ISSUE #29: [🧑‍💻] Testing : a11y testing
😊  Should be a11y compliant.

### LINKED ISSUE #35: [🧑‍💻] Testing : Instrument Code & Compile Code Coverage Report
[Instrument](https://docs.cypress.io/guides/tooling/code-coverage#Introduction) code using the [instrument-cra](https://github.com/cypress-io/instrument-cra) module:
`npm i @datlas/instrument-cra`

update package.json
```
scripts : { 
  "start": "react-scripts -r @datlas/instrument-cra start",
}
```
Screen Shot 2022-02-12 at 00 13 14

Insure your application has `__coverage__` in the dev tools console.  
`npm run dev` (using the scripts from the screenshot)
Once tests have completed, view the coverage report:
`open coverage/lcov-report/index.html`

### PR DESCRIPTION
**Pull request by elisebeall**
## [Mi Voz, Mi Voto] 🗳

- [ ] - Feature PR [🧩]
- [x] - Refactor PR [⚙️]
- [ ] - BugFix PR [🐞]
- [ ] - User Interface PR [🖍]
- [ ] - Error Handling PR [⚠️]
- [x] - Testing PR [🧑‍💻]

**What (if any) features are you implementing?**
 - Complete E2E testing of the email notification form and all error handling & loading.
 - Achieved 100% test coverage using the instrument-cra module & cypress code-coverage package
 
Screen Shot 2022-02-12 at 01 58 03
 
 - Closes #20 
 - Closes #24 
 - Closes #29 
 - Closes #35 

**What (if anything) did you refactor?**
 - The way error handling was approached needed to be refactored.  We needed to use the message the server sent back, both on a successful request, as well as an unsuccessful request.  Initially, the unsuccessful request would throw an error, which caused the Error component to load.  When the user tried to subscribe with a duplicate email address, the response object included {ok: false, status: 400} but it also had {body: {success}} or {body: {error}} and the success and error messages were very important info!
 - Closes #26 
 
**Were there any issues that arose?**
 - Stubbing out the entire request AND response cycle definitely had me tripped up.  I experimented with stubbing just the requests, and then testing the responses, but I think that will cause issues when continuous integration is implemented.  In the future, I would like to learn more about the use cases for stubbing either/or.
 - Instrumenting code is tricky!  I installed the cypress-coverage package which requires your code to be instrumented.  Installed the Istanbul package, which uses the nyc command line to to do the instrumentation.  Currently cypress coverage does not see the instrumented code.
 
**Is there anything needs to be prioritized/errors to be addressed?**
 - Currently, the test for accessible use of the form is being skipped.  Accessing the dropdown component and selecting the correct state using keyboard only commands has me stumped.

**Goals? Next steps?**
 - ~Implement #29~
 - Add a button to the error component that will return the user to the form.  Test that button!
 - ~Figure out how to use cypress coverage and nyc's instrumented code to get a coverage percentage.~

**⬇︎ Current Screenshots ⬇︎**
Screen Shot 2022-02-10 at 23 08 03

EDITED 2022.02.11 TO ADD MOST CURRENT SCREENSHOT:
Screen Shot 2022-02-12 at 01 47 56

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (ivonne-hernandez) [inline cypress.json:1]: I wasn't able to find the test that is using this file. If it's not needed, what do you think about removing it?

File: cypress/integration/email-notification-form-spec.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
+describe('Mi Voz, Mi Voto email notification form user flow', () => {
+  beforeEach(() => {
+    //cy.checkPageA11y('http://localhost:3000')
```
Comment (ivonne-hernandez) [inline cypress/integration/email-notification-form-spec.js]: What do you think about removing the commented out line 3?

File: cypress/integration/email-notification-form-spec.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        .focused().should('have.id', 'agree_to_emails')
+        .get('input[id=agree_to_emails]').should('have.value', 'false')
+        .realPress('Space')
+        .get('input[id=agree_to_emails]').should('have.value', 'true')
+        .realPress('Tab')
+        .realPress('Enter')
+        .wait('@successfulPost')
+        .get('.server-message').should('contain', 'You are now registered to receive notifications about upcoming elections in your state.')
+    })
+  });
```
Comment (ivonne-hernandez) [inline cypress/integration/email-notification-form-spec.js]: What do you think about removing this test since it's being skipped?

File: cypress/integration/email-notification-form-spec.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        .get('input[id=agree_to_emails]').click()
+        .get('.submit-button').click()
+        .wait('@getServerFailure')
+    })
+    cy.get('.error-text').should('contain', 'We\'re sorry, please try again.')
+  });
+
+  xit('Should display a button to send the user back to the form', () => {
+
+  });
```
Comment (ivonne-hernandez) [inline cypress/integration/email-notification-form-spec.js]: What do you think about removing this wireframe?

File: cypress/support/commands.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      })
+    })
+  })
+}
+
+Cypress.Commands.add('checkPageA11y', (path) => {
+  cy.visit(path);
+  cy.injectAxe();
+  cy.checkA11y(null, null, callback);
+})
```
Comment (ivonne-hernandez) [inline cypress/support/commands.js:61]: Is this file needed since we'll be removing the .checkPageA11y  comment?

File: instrumented/App.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    return (
+      <div className="App">
+        <Header />
+        <MainContainer />
+      </div>
+    );
+  }
+}
+
+export default App;
```
Comment (ivonne-hernandez) [inline instrumented/App.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/EmailNotificationForm/EmailNotificationForm.js [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+              </form>
+            }
+          </>
+        }
+      </>
+    );
+  }
+}
+
+export default EmailNotificationForm;
```
Comment (ivonne-hernandez) [inline instrumented/Components/EmailNotificationForm/EmailNotificationForm.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/EmailNotificationForm/states.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
...TWV4aWNvXCIgfSxcbiAgICB7IHZhbHVlOiBcIm55XCIsIGxhYmVsOiBcIk5ldyBZb3JrXCIgfSxcbiAgICB7IHZhbHVlOiBcIm5jXCIsIGxhYmVsOiBcIk5vcnRoIENhcm9saW5hXCIgfSxcbiAgICB7IHZhbHVlOiBcIm5kXCIsIGxhYmVsOiBcIk5vcnRoIERha290YVwiIH0sXG4gICAgeyB2YWx1ZTogXCJvaFwiLCBsYWJlbDogXCJPaGlvXCIgfSxcbiAgICB7IHZhbHVlOiBcIm9rXCIsIGxhYmVsOiBcIk9rbGFob21hXCIgfSxcbiAgICB7IHZhbHVlOiBcIm9yXCIsIGxhYmVsOiBcIk9yZWdvblwiIH0sXG4gICAgeyB2YWx1ZTogXCJwYVwiLCBsYWJlbDogXCJQZW5uc3lsdmFuaWFcIiB9LFxuICAgIHsgdmFsdWU6IFwicmlcIiwgbGFiZWw6IFwiUmhvZGUgSXNsYW5kXCIgfSxcbiAgICB7IHZhbHVlOiBcInNjXCIsIGxhYmVsOiBcIlNvdXRoIENhcm9saW5hXCIgfSxcbiAgICB7IHZhbHVlOiBcInNkXCIsIGxhYmVsOiBcIlNvdXRoIERha290YVwiIH0sXG4gICAgeyB2YWx1ZTogXCJ0blwiLCBsYWJlbDogXCJUZW5uZXNzZWVcIiB9LFxuICAgIHsgdmFsdWU6IFwidHhcIiwgbGFiZWw6IFwiVGV4YXNcIiB9LFxuICAgIHsgdmFsdWU6IFwidXRcIiwgbGFiZWw6IFwiVXRhaFwiIH0sXG4gICAgeyB2YWx1ZTogXCJ2dFwiLCBsYWJlbDogXCJWZXJtb250XCIgfSxcbiAgICB7IHZhbHVlOiBcInZhXCIsIGxhYmVsOiBcIlZpcmdpbmlhXCIgfSxcbiAgICB7IHZhbHVlOiBcIndhXCIsIGxhYmVsOiBcIldhc2hpbmd0b25cIiB9LFxuICAgIHsgdmFsdWU6IFwid3ZcIiwgbGFiZWw6IFwiV2VzdCBWaXJnaW5pYVwiIH0sXG4gICAgeyB2YWx1ZTogXCJ3aVwiLCBsYWJlbDogXCJXaXNjb25zaW5cIiB9LFxuICAgIHsgdmFsdWU6IFwid3lcIiwgbGFiZWw6IFwiV3lvbWluZ1wiIH1cbiAgXVxufSJdfQ==
```
Comment (ivonne-hernandez) [inline instrumented/Components/EmailNotificationForm/states.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/Error/Error.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    <div className="error-container">
+      <img className="error-image" src={noVoice1}
+        alt="microphone with horizontal strikethrough" />
+      <h3 className="error-text">We're sorry, please try again.</h3>
+      <h4 className="error-text">{error}</h4>
+    </div>
+  )
+}
+
+export default Error;
```
Comment (ivonne-hernandez) [inline instrumented/Components/Error/Error.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/Header/Header.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      </button>
+      <button
+        className="en-espanol-button header-button">
+        En Español
+      </button>
+    </div>
+  );
+}
+
+export default Header;
```
Comment (ivonne-hernandez) [inline instrumented/Components/Header/Header.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/Loading/Loading.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+const Loading = () => {
+  return (
+    <div className="loading-icon-container">
+      <img className="loading-icon" src={loading} alt="loading icon" />
+    </div>
+  )
+}
+
+export default Loading;
```
Comment (ivonne-hernandez) [inline instrumented/Components/Loading/Loading.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/Components/MainContainer/MainContainer.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+import EmailNotificationForm from '../EmailNotificationForm/EmailNotificationForm';
+
+const MainContainer = () => {
+  return (
+    <main className="main-container">
+      <EmailNotificationForm />
+    </main>
+  );
+}
+export default MainContainer;
```
Comment (ivonne-hernandez) [inline instrumented/Components/MainContainer/MainContainer.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/apiCalls.js [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
...ZXdFbWFpbFN1YnNjcmliZXIiLCJuZXdFbWFpbFN1YnNjcmliZXIiLCJmZXRjaCIsInVzZXJzIiwibW9kZSIsIm1ldGhvZCIsImhlYWRlcnMiLCJib2R5IiwiSlNPTiIsInN0cmluZ2lmeSJdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0FBZVk7Ozs7Ozs7OztBQWZaLE9BQU9BLFNBQVAsTUFBc0IsZ0JBQXRCOztBQUVBLE9BQU8sTUFBTUMsc0JBQXNCLEdBQUlDLGtCQUFELElBQXdCO0FBQUE7QUFBQTtBQUM1RCxTQUFPQyxLQUFLLENBQUNILFNBQVMsQ0FBQ0ksS0FBWCxFQUFrQjtBQUM1QkMsSUFBQUEsSUFBSSxFQUFFLE1BRHNCO0FBRTVCQyxJQUFBQSxNQUFNLEVBQUUsTUFGb0I7QUFHNUJDLElBQUFBLE9BQU8sRUFBRTtBQUNQLHNCQUFnQixrQkFEVDtBQUVQLGdCQUFVO0FBRkgsS0FIbUI7QUFPNUJDLElBQUFBLElBQUksRUFBRUMsSUFBSSxDQUFDQyxTQUFMLENBQWVSLGtCQUFmO0FBUHNCLEdBQWxCLENBQVo7QUFTRCxDQVZNIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IGVuZHBvaW50cyBmcm9tICcuL2VuZHBvaW50cy5qcyc7XG5cbmV4cG9ydCBjb25zdCBwb3N0TmV3RW1haWxTdWJzY3JpYmVyID0gKG5ld0VtYWlsU3Vic2NyaWJlcikgPT4ge1xuICByZXR1cm4gZmV0Y2goZW5kcG9pbnRzLnVzZXJzLCB7XG4gICAgbW9kZTogJ2NvcnMnLFxuICAgIG1ldGhvZDogJ1BPU1QnLFxuICAgIGhlYWRlcnM6IHtcbiAgICAgICdDb250ZW50LVR5cGUnOiAnYXBwbGljYXRpb24vanNvbicsXG4gICAgICAnQWNjZXB0JzogJ2FwcGxpY2F0aW9uL2pzb24nXG4gICAgfSxcbiAgICBib2R5OiBKU09OLnN0cmluZ2lmeShuZXdFbWFpbFN1YnNjcmliZXIpXG4gIH0pXG59XG4iXX0=
```
Comment (ivonne-hernandez) [inline instrumented/apiCalls.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/endpoints.js [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
...Y3JpYmVyLiBBIHZhbGlkIGVtYWlsIG11c3QgYmUgcHJvdmlkZWQuXCIsXG4gICAgICAgIHN0YXR1czogNDAwXG4gICAgICB9XG4gICAgICBFTUFJTCBBTFJFQURZIEVYSVNUUzpcbiAgICAgIHtcbiAgICAgICAgZXJyb3I6IFwiI3t1c2VyLmVtYWlsfSBpcyBhbHJlYWR5IHN1c2JzY3JpYmVkIHRvIHJlY2VpdmUgZWxlY3Rpb24gbm90aWZpY2F0aW9ucy5cIixcbiAgICAgICAgc3RhdHVzOiA0MDBcbiAgICAgIH1cbiAgICAgIEdFTkVSQUwgRVJST1I6XG4gICAgICB7XG4gICAgICAgIGVycm9yOiBcIkVycm9yIGNyZWF0aW5nIHN1YnNjcmliZXIuIFBsZWFzZSBlbnN1cmUgYWxsIGVsZW1lbnRzIGluIHRoZSBmb3JtIGFyZSBmaWxsZWQgb3V0IGNvcnJlY3RseS5cIixcbiAgICAgICAgc3RhdHVzOiA0MDBcbiAgICAgIH1cbiAgICAgIFNVQ0NFU1NGVUxMWSBSRU1PVkVEOlxuICAgICAge1xuICAgICAgICBzdWNjZXNzOiBcIllvdSBoYXZlIHN1Y2Nlc3NmdWxseSB1bnN1YnNjcmliZWQgZnJvbSBNaSBWb3RvLCBNaSBWb3oncyBlbWFpbCBsaXN0LiBZb3Ugd2lsbCBubyBsb25nZXIgcmVjZWl2ZSBlbWFpbCBub3RpZmljYXRpb25zIGF0ICN7dXNlci5lbWFpbH0uXCJ9LFxuICAgICAgICBzdGF0dXM6IDIwMFxuICAgICAgfVxuICAqL1xuICBlbGVjdGlvbnM6ICdodHRwOi8vbG9jYWxob3N0OjMwMDEvYXBpL3YxL2VsZWN0aW9ucycsXG4gIC8qXG4gICAgUVVFUlkgUEFSQU1TPyBMT0NBVElPTj8gU1RBVEU/ICBJLkUuIENPP1xuICAqL1xuICByZWZlcmVuZHVtczogJ2h0dHA6Ly9sb2NhbGhvc3Q6MzAwMS9hcGkvdjEvcmVmZXJlbmR1bXMnLFxuICBzdGF0ZTogJ2h0dHA6Ly9sb2NhbGhvc3Q6MzAwMS9hcGkvdjEvc3RhdGUnXG59XG5cbmV4cG9ydCBkZWZhdWx0IGVuZHBvaW50cztcbiJdfQ==
```
Comment (ivonne-hernandez) [inline instrumented/endpoints.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: instrumented/index.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root')
+);
```
Comment (ivonne-hernandez) [inline instrumented/index.js]: I didn't see this file being used anywhere in the tests. Can you please explain what this file is for and if it's needed?

File: package.json [hunk 16]
```diff
@@ -14,7 +15,9 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "start-server-and-test start http://localhost:3000 cy:run",
```
Comment (ivonne-hernandez) [inline package.json]: Can you please explain how this script works and the idea behind setting it up like this?

**Review by ivonne-hernandez — CHANGES_REQUESTED** [review]
Thanks for taking on testing for the form Elise. I've outlined a few requested changes in the comments prior to having this branch merged into main.

File: cypress.json (hunk 1 above)
Comment (elisebeall) [reply cypress.json:1]: This is something that Cypress adds on install.  I believe it is needed to run the package.

File: cypress/integration/email-notification-form-spec.js (hunk 2 above)
Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: So, by uncommenting that line (and commenting out the cy.visit), cypress runs accessibility tests!
Screen Shot 2022-02-11 at 10 15 12
Screen Shot 2022-02-11 at 10 16 00

Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: `$Form elements must have labels` is in reference to the language radio buttons.
`$Page should have a level-one heading` is in reference to the fact that the title of the site is wrapped in a `<p>` tag, and it should be an `<h1>` 
`$All page content should be contained by landmarks` ...  Well I wasn't sure about this one, but upon clicking the link, it sends me [here](https://dequeuniversity.com/rules/axe/4.4/region?application=axeAPI).  It looks the outtermost `div` in our Header component should be changed to a semantic `<header>` instead.  At least that is my guess.

Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: @ivonne-hernandez Would you like me to make these changes on this branch, before the merge?

File: cypress/integration/email-notification-form-spec.js (hunk 3 above)
Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: I am going to get this to work gosh darn it!  It's funny, it actually works about half of the time.  When Cypress gets to the `state_name` select, it almost seems like it doesn't wait long enough to get 'co'.  I tried adding a `cy.wait` at some point, but things changed after that.  Maybe I will try it again...

File: cypress/integration/email-notification-form-spec.js (hunk 4 above)
Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: I am going to leave it for now, as it will be used once router is implemented and a button can be added.  I'm sure my brain won't let me forget about it, but it's one more line of defense against forgetting to test that functionality. 😅

File: cypress/support/commands.js (hunk 5 above)
Comment (elisebeall) [reply cypress/support/commands.js:61]: Yes, this file is a default Cypress file and will break it if we remove it.

File: instrumented/Components/EmailNotificationForm/EmailNotificationForm.js (hunk 7 above)
Comment (elisebeall) [reply instrumented/Components/EmailNotificationForm/EmailNotificationForm.js]: Instrumented files are needed for coverage tests.  Essentially, instrumenting adds numbering to the code so when the cypress-coverage package is ran, it can evaluate whether or not a line was run by the tests.  At least that's my basic understanding of it!

File: instrumented/Components/EmailNotificationForm/states.js (hunk 8 above)
Comment (elisebeall) [reply instrumented/Components/EmailNotificationForm/states.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/Components/MainContainer/MainContainer.js (hunk 12 above)
Comment (elisebeall) [reply instrumented/Components/MainContainer/MainContainer.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/Components/Loading/Loading.js (hunk 11 above)
Comment (elisebeall) [reply instrumented/Components/Loading/Loading.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/Components/Header/Header.js (hunk 10 above)
Comment (elisebeall) [reply instrumented/Components/Header/Header.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/apiCalls.js (hunk 13 above)
Comment (elisebeall) [reply instrumented/apiCalls.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/endpoints.js (hunk 14 above)
Comment (elisebeall) [reply instrumented/endpoints.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: package.json (hunk 16 above)
Comment (elisebeall) [reply package.json]: This is used to run cypress headless as explained [here](https://docs.cypress.io/guides/continuous-integration/introduction#Boot-your-server).  Running cypress headless is needed for continuous integration.

File: instrumented/index.js (hunk 15 above)
Comment (elisebeall) [reply instrumented/index.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/Components/Error/Error.js (hunk 9 above)
Comment (elisebeall) [reply instrumented/Components/Error/Error.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

File: instrumented/App.js (hunk 6 above)
Comment (elisebeall) [reply instrumented/App.js]: 'Explained' [here](https://github.com/mi-voz-mi-voto/mi-voz-mi-voto/pull/30#discussion_r804866273)

**Comment by elisebeall** [comment]
@ivonne-hernandez I read through your comments and am making changes now.  Please review my replies and let me know what you think.  Thanks!

File: cypress/integration/email-notification-form-spec.js (hunk 2 above)
Comment (ivonne-hernandez) [reply cypress/integration/email-notification-form-spec.js]: Absolutely, please move forward with making those changes. I would just ensure that the styling isn't impacted as you refactor each of these tags.

File: cypress/integration/email-notification-form-spec.js (hunk 3 above)
Comment (ivonne-hernandez) [reply cypress/integration/email-notification-form-spec.js]: We can always come back and add more tests like this one. For now, I think it would be good to remove it since we don't want it to interfere with our ability to implement Continuous Integration and Deployment.

File: cypress/integration/email-notification-form-spec.js (hunk 4 above)
Comment (ivonne-hernandez) [reply cypress/integration/email-notification-form-spec.js]: We can always come back and add more tests like this one. For now, I think it would be good to remove it since we don't want it to interfere with our ability to implement Continuous Integration and Deployment.

**Comment by ivonne-hernandez** [comment]
Here are some icons we can use for the success/failure message:
green check
red x

File: cypress/integration/email-notification-form-spec.js (hunk 3 above)
Comment (elisebeall) [reply cypress/integration/email-notification-form-spec.js]: Got it!

Screen Shot 2022-02-11 at 13 25 59

**Comment by elisebeall** [comment]
Changes have been implemented to html & JSX to close #29

File: cypress/fixtures/success.json [hunk 17]
```diff
@@ -0,0 +1,3 @@
+{
+  "success": "A confirmation email for state election reminders has been sent to "
+}
```
Comment (ivonne-hernandez) [inline cypress/fixtures/success.json:3]: Once our BE is refactored to return this statement, will this fixture need to be refactored as well to include the email address that was submitted?

**Review by ivonne-hernandez — APPROVED** [review]
This branch is ready to be merged into main.
